include/api.php

```diff
@@ -237,9 +237,13 @@ function api_login(App $a)
 	if ($addon_auth['authenticated'] && !empty($addon_auth['user_record'])) {
 		$record = $addon_auth['user_record'];
 	} else {
-		$user_id = User::authenticate(trim($user), trim($password), true);
-		if ($user_id !== false) {
-			$record = DBA::selectFirst('user', [], ['uid' => $user_id]);
+		try {
+			$user_id = User::getIdFromPasswordAuthentication(trim($user), trim($password), true);
+			if ($user_id !== false) {
+				$record = DBA::selectFirst('user', [], ['uid' => $user_id]);
+			}
+			} catch (Exception $ex) {
+				$record = [];
 		}
```

Comment:
`User::getIdFromPasswordAuthentication` will never return something else than a successful authenticated user id. All other cases will throw an Exception, either NotFound if user doesn't exist or Forbidden if credentials were wrong.
```suggestion
		try {
			$user_id = User::getIdFromPasswordAuthentication(trim($user), trim($password), true);
			$record = DBA::selectFirst('user', [], ['uid' => $user_id]);
		} catch (Exception $ex) {
			$record = [];
		}
```